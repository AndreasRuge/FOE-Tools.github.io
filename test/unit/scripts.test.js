describe("Scripts", () => {
  var normalizedPath = require("path").join(__dirname, "scripts");

  require("fs")
    .readdirSync(normalizedPath)
    .forEach(function(file) {
      require("./scripts/" + file);
    });
});
