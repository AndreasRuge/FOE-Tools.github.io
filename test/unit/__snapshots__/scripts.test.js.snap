// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scripts Promotion Message Builder Valid value with "All variables" 1`] = `
"Foo foe_data.gb_short.Observatory from 9 to 10 (430|650):
<<
9—10 'foe_data.gb_short.Observatory' 4_10(300)
>>
<<
9—10 'foe_data.gb_short.Observatory' 3_30(240)
>>
<<
9—10 'foe_data.gb_short.Observatory' 2_100(220)
>>
<<
9—10 'foe_data.gb_short.Observatory' 1_185(200)
>>
And custom place P1(185|200), P2(100|220), P3(30|240), P4(10|300), P5(|) Bar"
`;

exports[`Scripts Promotion Message Builder Valid value with "Custom 0" 1`] = `"Yoratheon foe_data.gb.Observatory {650} 🆙 4 (10) 👈 3 (30) 👈 2 (100) 👈 1 (185)"`;

exports[`Scripts Promotion Message Builder Valid value with "Custom 11" 1`] = `"foe_data.gb.Observatory 9 < 4,3,2,1 > 10"`;

exports[`Scripts Promotion Message Builder Valid value with "Default 1" 1`] = `"Your pseudo foe_data.gb.Observatory 9 → 10 P1(185) P2(100) P3(30) P4(10) up in half hour"`;

exports[`Scripts Promotion Message Builder Valid value with "Default 2" 1`] = `"Yoratheon P1(185) P2(100) P3(30) P4(10) foe_data.gb.Observatory 9 → 10 up dans 30 min"`;

exports[`Scripts Promotion Message Builder Valid value with "Default 3" 1`] = `"Your pseudo foe_data.gb.Observatory 9 → 10 P4(10) P3(30) P2(100) P1(185) up in half hour"`;

exports[`Scripts Promotion Message Builder Valid value with "Default 4" 1`] = `"Yoratheon P4(10) P3(30) P2(100) P1(185) foe_data.gb.Observatory 9 → 10 up dans 30 min"`;

exports[`Scripts Promotion Message Builder Valid value with "Default 5" 1`] = `"Yoratheon foe_data.gb.Observatory 9 → 10 1 2 3 4 up dans 30 min"`;

exports[`Scripts Promotion Message Builder Valid value with "Default 6" 1`] = `"Yoratheon foe_data.gb.Observatory 9 → 10 4 3 2 1 up dans 30 min"`;

exports[`Scripts Promotion Message Builder Valid value with "Default 7" 1`] = `"псевдонимы foe_data.gb.Observatory 9 → 10 1,2,3,4"`;

exports[`Scripts Promotion Message Builder Valid value with "Default 8" 1`] = `"إسم مستعار foe_data.gb.Observatory 9 → 10 4,3,2,1"`;

exports[`Scripts Promotion Message Builder Valid value with "Default 11" 1`] = `
" - foe_data.gb.Observatory 1
 - foe_data.gb.Observatory 2
 - foe_data.gb.Observatory 3
 - foe_data.gb.Observatory 4"
`;

exports[`Scripts Utils handlerForm Valid value and save in cookie 1`] = `
Array [
  Object {
    "key": "profile/myKey",
    "value": 5,
  },
]
`;

exports[`Scripts Utils handlerForm Valid value and save in cookie with key 'foo' 1`] = `
Array [
  Object {
    "key": "profile/foo",
    "value": 5,
  },
]
`;

exports[`Scripts Utils handlerForm Valid value with float value and save in cookie with path '/foo' 1`] = `
Array [
  Object {
    "key": "profile/foo",
    "value": 5.3,
  },
]
`;
